src/main.ts

```diff
@@ -9,9 +9,15 @@ const cpExec = util.promisify(require('child_process').exec);
 import { createScriptFile, TEMP_DIRECTORY, NullOutstreamStringWritable, deleteFile, getCurrentTime, checkIfEnvironmentVariableIsOmitted } from './utils';
 
 const START_SCRIPT_EXECUTION_MARKER: string = "Starting script execution via docker image mcr.microsoft.com/azure-cli:";
-const AZ_CLI_VERSION_DEFAULT_VALUE = 'agentazcliversion'
+const AZ_CLI_VERSION_DEFAULT_VALUE = 'agentazcliversion';
+const prefix = !!process.env.AZURE_HTTP_USER_AGENT ? `${process.env.AZURE_HTTP_USER_AGENT}` : "";
 
 export async function main() {
+    let usrAgentRepo = `${process.env.GITHUB_REPOSITORY}`;
+    let actionName = 'AzureCLIAction';
+    let userAgentString = (!!prefix ? `${prefix}+` : '') + `GITHUBACTIONS/${actionName}_${usrAgentRepo}`;

```

Comment:
From my understanding, the logic here appends `GITHUBACTIONS/${actionName}_${usrAgentRepo}` to existing `AZURE_HTTP_USER_AGENT`:

```
$AZURE_HTTP_USER_AGENT GITHUBACTIONS/${actionName}_${usrAgentRepo}
```

Why not prefixing `GITHUBACTIONS/${actionName}_${usrAgentRepo}` to `AZURE_HTTP_USER_AGENT` instead:

```
GITHUBACTIONS/${actionName}_${usrAgentRepo} $AZURE_HTTP_USER_AGENT
```

Reply:
A normal `User-Agent` looks like:

```
> az group show --name cli_test_azure_firewall_rules_with_ipgroupse4upfibqoujmdn4odvtgrbuvqqgdvwd7 --debug
...
cli.azure.cli.core.sdk.policies:     'User-Agent': 'AZURECLI/2.53.0 (MSI) azsdk-python-azure-mgmt-resource/23.1.0b2 Python/3.10.10 (Windows-10-10.0.22635-SP0)'
```

Reply:
Both are ok. I just want to align with azure/login & azure/powershell.

Reply:
Is the plus sign in `${prefix}+` literal?

Reply:
> Is the plus sign in `${prefix}+` literal?

Yes. Please see https://github.com/Azure/cli/pull/122#issuecomment-1821994298 for detailed example.